Enable Asteroids to Spawn Health Packs when Ship Health is Not Full

rajnpatel opened the pull request:
Had to make int shipHealth public so AsteroidMovement.cs could access it. Added coroutine in AsteroidMovement.cs so health packs can't spawn with the bullets power up simultaneously.

rajnpatel committed:
Added health packs for asteroids to spawn when ship health is not full

Had to make int shipHealth public so AsteroidMovement.cs could access it. Added coroutine in AsteroidMovement.cs so health packs can't spawn with the bullets power up simultaneously.

adamlbunch reviewed (changes requested):
Revert Level1.unity pls

adamlbunch commented on Assets/Scripts/Power-ups/HealthPack.cs:
```diff
@@ -0,0 +1,31 @@
+    {
+        Destroy(gameObject);
+    }
+}
```
Fix spacing throughout the file

adamlbunch commented on Assets/Scripts/Asteroid/AsteroidMovement.cs:
```diff
@@ -26,16 +27,31 @@ private void OnCollisionEnter2D(Collision2D col)
+    private IEnumerator NoAppear()
+    {
+        yield return new WaitForSeconds(1.0f);
+        canAppear = true;
```
Fix spacing and formatting

adamlbunch commented on Assets/Scripts/Asteroid/AsteroidMovement.cs:
```diff
@@ -26,16 +27,31 @@ private void OnCollisionEnter2D(Collision2D col)
+                    canAppear = false;
+                    StartCoroutine(NoAppear());
+                }
+                if (Random.Range(1.00f, 2.00f) < 1.10 && shipHealth < 10 && canAppear)
```
Why are you using random range between 1 and 2? I think you can use 0 to 1 here as well.

adamlbunch commented on Assets/Scripts/Asteroid/AsteroidMovement.cs:
```diff
@@ -26,16 +27,31 @@ private void OnCollisionEnter2D(Collision2D col)
             {
                 Destroy(gameObject);
+
                 if (Random.Range(0.00f, 1.00f) < 0.25)
```
Let's turn 0.25 into a public variable that can be adjusted. Hardcoded values are bad. Please name appropriately.

adamlbunch commented on Assets/Scripts/Asteroid/AsteroidMovement.cs:
```diff
@@ -26,16 +27,31 @@ private void OnCollisionEnter2D(Collision2D col)
+
+    private IEnumerator NoAppear()
+    {
+        yield return new WaitForSeconds(1.0f);
```
Turn 1.0f into a public variable that can be adjusted. No hardcoded values

rajnpatel committed:
Fixed formatting and spacing. Added variables in lieu of hard coded values.

adamlbunch commented on Assets/Scripts/Asteroid/AsteroidMovement.cs:
```diff
@@ -50,12 +54,12 @@ private void OnCollisionEnter2D(Collision2D col)
     private IEnumerator NoAppear()
     {
-        yield return new WaitForSeconds(1.0f);
+        yield return new WaitForSeconds(delayPowerUp);
```
rename to powerUpDelay

rajnpatel committed:
Renamed variable

rajnpatel committed:
Changed delayTime of PlayerShoot

The delay time of PlayerShoot needs to remain the same with its default value and when hit by something.

adamlbunch committed:
Merge branch 'develop' into Enable-Asteroids-to-Spawn-Health-Packs-when-Ship-Health-is-Not-Full
